cmd/run.go

```diff
@@ -226,26 +226,26 @@ type cmdFn func(v *viper.Viper) (int, error)
 
 // RunCmd runs a command function and exits with the exit code returned by
 // the command function. Will send diagnostic on any errors or panics.
-func RunCmd(v *viper.Viper, verbose bool, cmd cmdFn) {
-	exitCode := runCmd(v, verbose, cmd)
+func RunCmd(v *viper.Viper, params *logfile.Params, cmd cmdFn) {
+	exitCode := runCmd(v, params, cmd)
 
 	os.Exit(exitCode)
 }
 
 // RunCmdWithOfflineSync runs a command function and exits with the exit code
 // returned by the command function. If command run was successful, it will execute
 // offline sync command afterwards. Will send diagnostic on any errors or panics.
-func RunCmdWithOfflineSync(v *viper.Viper, verbose bool, cmd cmdFn) {
-	exitCode := runCmd(v, verbose, cmd)
+func RunCmdWithOfflineSync(v *viper.Viper, params *logfile.Params, cmd cmdFn) {
+	exitCode := runCmd(v, params, cmd)
 	if exitCode != exitcode.Success {
 		os.Exit(exitCode)
 	}
 
-	os.Exit(runCmd(v, verbose, offlinesync.Run))
+	os.Exit(runCmd(v, params, offlinesync.Run))
 }
 
 // runCmd contains the main logic of RunCmd.
-func runCmd(v *viper.Viper, verbose bool, cmd cmdFn) int {
+func runCmd(v *viper.Viper, params *logfile.Params, cmd cmdFn) int {
```

Comment:
Instead passing the whole object it's important to keep the function signature clear what's doing under the hood. It's better to pass `Verbose` and `SendDiagsOnErrors` only.

```suggestion
func runCmd(v *viper.Viper, verbose,  sendDiagsOnErrors bool, cmd cmdFn) int {
```

Reply:
I don't think having those params helps much... either way the reader has to go into `runCmd` to see what it's doing right?

Reply:
This style guide even says to use a custom type instead of passing multiple primitive variables https://github.com/uber-go/guide/blob/master/style.md#avoid-naked-parameters

Reply:
A struct would make it much easier to read, but this func will have only 4 input params, which is acceptable.